core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -147,11 +147,8 @@ trait EntityEncoderInstances extends EntityEncoderInstances0 {
   implicit val byteArrayEncoder: EntityEncoder[Array[Byte]] =
     chunkEncoder.contramap(Chunk.bytes)
 
-  // TODO fs2 port this is where we miss ByteVector
-  /*
   implicit val byteBufferEncoder: EntityEncoder[ByteBuffer] =
-    chunkEncoder.contramap(ByteVector.view)
-   */
+    chunkEncoder.contramap(bb => Chunk.bytes(bb.array))
```

Comment:
`.array` is not a guaranteed operation.  I don't do enough raw nio to know in what cases it's not supported.

scodec-bits uses a [view of a slice](https://github.com/scodec/scodec-bits/blob/0abe269dfda050cc75d27ed384e3e3a496445971/core/shared/src/main/scala/scodec/bits/ByteVector.scala#L1479-L1482).  A `Chunk[Byte]` could be invented for similar effect, but it would have very poor concatenation performance.

This might be a good design question for the fs2 experts.